core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -133,19 +134,25 @@ public void execute(final Operation operation, final User user) throws Operation
      * @throws OperationException thrown if the job fails to run.
      */
     public JobDetail executeJob(final OperationChain<?> operationChain, final User user) throws OperationException {
-        updateOperationChainView(operationChain);
+        try {
+            updateOperationChainView(operationChain);
 
-        for (final GraphHook graphHook : graphHooks) {
-            graphHook.preExecute(operationChain, user);
-        }
+            for (final GraphHook graphHook : graphHooks) {
+                graphHook.preExecute(operationChain, user);
+            }
 
-        JobDetail result = store.executeJob(operationChain, user);
+            JobDetail result = store.executeJob(operationChain, user);
 
-        for (final GraphHook graphHook : graphHooks) {
-            result = graphHook.postExecute(result, operationChain, user);
-        }
+            for (final GraphHook graphHook : graphHooks) {
+                result = graphHook.postExecute(result, operationChain, user);
+            }
 
-        return result;
+            return result;
+
+        } catch (final Exception e) {
```

Comment:
We need to ensure the operation inputs are always closed when any exception is thrown, otherwise we risk leaving connections to accumulo/hbase open.